modules/ipc/include/hephaestus/ipc/zenoh/service_client.h

```diff
@@ -0,0 +1,46 @@
+//================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <chrono>
+#include <optional>
+#include <utility>
+#include <vector>
+
+#include "hephaestus/ipc/topic.h"
+#include "hephaestus/ipc/zenoh/service.h"
+#include "hephaestus/ipc/zenoh/session.h"
+
+namespace heph::ipc::zenoh {
+
+template <typename RequestT, typename ReplyT>
+class ServiceClient {
+public:
+  ServiceClient(SessionPtr session, TopicConfig topic_config);
+  auto call(const RequestT& request, const std::optional<std::chrono::milliseconds>& timeout = std::nullopt)
```

Comment:
For robustness reasons, I would not make the timeout optional. The user should always provide a timeout, and reaching the timeout should always notify, log, except.

No timeout is a dead lock.

Reply:
Good point, yes